### PR TITLE
Refactor model to use yaramo model

### PR DESCRIPTION
**Pull request by arneboockmeyer**

### CODE REVIEW

File: infrastructureprovider/infrastructureprovider.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    @abstractmethod
+    def set_signal_state(self, yaramo_signal, target_state):
+        pass
+
+    #
+    # Train Detection System
+    #
+
+    def set_tds_count_in_callback(self, tds_count_in_callback):
+        self.tds_count_in_callback = tds_count_in_callback
```
Comment (SaturnHafen) [inline infrastructureprovider/infrastructureprovider.py:31]: This method should probably be marked private, as it is used for *notifying* the interlocking controller, right?

```suggestion
    def _set_tds_count_in_callback(self, tds_count_in_callback):
        self.tds_count_in_callback = tds_count_in_callback
```

File: infrastructureprovider/infrastructureprovider.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    def tds_count_in(self, track_segment_id):
+        self.tds_count_in_callback(track_segment_id)
+
+    def set_tds_count_out_callback(self, tds_count_out_callback):
+        self.tds_count_out_callback = tds_count_out_callback
```
Comment (SaturnHafen) [inline infrastructureprovider/infrastructureprovider.py:37]:
```suggestion
    def _set_tds_count_out_callback(self, tds_count_out_callback):
        self.tds_count_out_callback = tds_count_out_callback
```

File: infrastructureprovider/infrastructureprovider.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    def tds_count_out(self, track_segment_id):
+        self.tds_count_out_callback(track_segment_id)
```
Comment (SaturnHafen) [inline infrastructureprovider/infrastructureprovider.py:40]:
```suggestion
    def tds_count_out(self, track_segment_id):
        """Removes a train to the segment identified by the `segment_id`"""
        self.tds_count_out_callback(track_segment_id)
```

**Review by SaturnHafen — COMMENTED** [review]
I am not qualified to review the interlocking logic. But I personally found the documentation of the API lacking. Additionally we were unable to run the example and we were unable to import the project into our project. The latter can be fixed by providing a `pyproject.toml` which declares the dependencies and exports the components. This would probably fix our issues with running the project too, as it would pin the dependencies (in this case yaramo) to a specific version. We are suspecting that the reason for our issues is an outdated yaramo version.

**Comment by arneboockmeyer** [comment]
> I am not qualified to review the interlocking logic. But I personally found the documentation of the API lacking. Additionally we were unable to run the example and we were unable to import the project into our project. The latter can be fixed by providing a `pyproject.toml` which declares the dependencies and exports the components. This would probably fix our issues with running the project too, as it would pin the dependencies (in this case yaramo) to a specific version. We are suspecting that the reason for our issues is an outdated yaramo version.

What kind of issues do you have? Do you have some error messages? I just double checked and it works with the current main branch of the `yaramo` repository. Probably we havn't updated the yaramo version number for a while, so please uninstall and install yaramo again. 
`pyproject.toml` would be great - agree.

**Pull request merged**